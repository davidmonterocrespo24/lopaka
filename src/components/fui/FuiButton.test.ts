import {mount} from '@vue/test-utils';
import {describe, expect, it} from 'vitest';
import FuiButton from './FuiButton.vue';

describe('FuiButton', () => {
    it('renders a button element', () => {
        const wrapper = mount(FuiButton);
        expect(wrapper.find('span').exists()).toBe(true);
        expect(wrapper.element).toMatchSnapshot();
    });

    it('emits a click event when clicked', async () => {
        const wrapper = mount(FuiButton);
        await wrapper.trigger('click');
        expect(wrapper.emitted('click')).toBeTruthy();
    });
});
