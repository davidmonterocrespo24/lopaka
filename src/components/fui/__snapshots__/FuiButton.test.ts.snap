// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FuiButton > renders a button element 1`] = `
<span
  class="button"
>
  
  
</span>
`;
